DataGenerator/src/main/java/org/opensearch/migrations/data/WorkloadGenerator.java

```diff
@@ -43,11 +45,25 @@ public void generate(WorkloadOptions options) {
 
     private List<CompletableFuture<?>> generateDocs(String indexName, Workload workload, WorkloadOptions options) {
         // This happens inline to be sure the index exists before docs are indexed on it
-        client.createIndex(indexName, workload.createIndex(options.index.indexSettings.deepCopy()), null);
+        var indexRequestDoc = workload.createIndex(options.index.indexSettings.deepCopy());
+        log.atInfo().setMessage("Creating index {} with {}").addArgument(indexName).addArgument(indexRequestDoc).log();
+        client.createIndex(indexName, indexRequestDoc, null);
 
         var docIdCounter = new AtomicInteger(0);
         var allDocs = workload.createDocs(options.totalDocs)
-            .map(doc -> new DocumentReindexer.BulkDocSection(indexName + "_ " + docIdCounter.incrementAndGet(), doc.toString()))
+            .map(doc -> {
```

Comment:
I would have liked to have used peek() here, but I didn't want to have to lose the sonarqube check for the whole file.